fileconnector/fileconnector-2.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/FileArchives.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.connector;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import javax.xml.stream.XMLStreamException;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.vfs2.*;
+import org.apache.commons.vfs2.impl.StandardFileSystemManager;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.codehaus.jettison.json.JSONException;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.Connector;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+import org.wso2.carbon.connector.util.*;
+
+public class FileArchives extends AbstractConnector implements Connector {
+    private static final Log log = LogFactory.getLog(FileArchives.class);
+    private final byte[] bytes = new byte[FileConstants.BUFFER_SIZE];
+
+    public void connect(MessageContext messageContext) {
+        String source = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.FILE_LOCATION);
+        String destinstion = (String) ConnectorUtils.lookupTemplateParamater(messageContext,
+                FileConstants.NEW_FILE_LOCATION);
+        boolean resultStatus = fileCompress(messageContext, source, destinstion);
+        generateResults(messageContext, resultStatus);
+    }
+
+    /**
+     * @param messageContext The message context that is generated for processing the file
+     * @param source         The file to be archived
+     * @param destination    Destination of the archived file
+     * @return return status
```

Comment:
Could you explain when the status would be true ?

Reply:
https://github.com/wso2-extensions/esb-connector-file/pull/28